### PR TITLE
Feature/CVSB-3797

### PR DESCRIPTION
**Pull request by oalexandru**

### CODE REVIEW

File: src/functions/certGen.ts [hunk 1]
```diff
@@ -23,18 +23,23 @@ const certGen: Handler = async (event: any, context?: Context, callback?: Callba
 
     event.Records.forEach(async (record: any) => {
         const testResult: any = JSON.parse(record.body);
-        const generatedCertificateResponse: Promise<ManagedUpload.SendData> = certificateGenerationService.generateCertificate(testResult)
-        .then((response: IGeneratedCertificateResponse) => {
-            return certificateUploadService.uploadCertificate(response);
-        });
+        console.log(testResult.testResultId)
```
Comment (KHodderDVSA) [inline src/functions/certGen.ts]: Should this be here?

Comment (oalexandru) [reply src/functions/certGen.ts]: no. removed it

File: src/functions/certGen.ts [hunk 2]
```diff
@@ -23,18 +23,22 @@ const certGen: Handler = async (event: any, context?: Context, callback?: Callba
...
+            certificateUploadPromises.push(Promise.reject(new Error("Record does not have valid testResultId for certificate generation")));
+        }
     });
 
     return Promise.all(certificateUploadPromises)
-    .catch((error: AWSError) => {
-        console.error(error);
-    });
+        .catch((error: Error) => {
+            console.error(error);
```
Comment (acozianu) [inline src/functions/certGen.ts]: Should we throw the errors the same as I did on ATF Report?

File: src/functions/certGen.ts [hunk 3]
```diff
@@ -23,18 +23,22 @@ const certGen: Handler = async (event: any, context?: Context, callback?: Callba
...
+        if (testResult.testResultId.match("\\b[a-zA-Z0-9]{8}\\b-\\b[a-zA-Z0-9]{4}\\b-\\b[a-zA-Z0-9]{4}\\b-\\b[a-zA-Z0-9]{4}\\b-\\b[a-zA-Z0-9]{12}\\b")) {
+            const generatedCertificateResponse: Promise<ManagedUpload.SendData> = certificateGenerationService.generateCertificate(testResult)
+                .then((response: IGeneratedCertificateResponse) => {
+                    return certificateUploadService.uploadCertificate(response);
+                });
 
-        certificateUploadPromises.push(generatedCertificateResponse);
+            certificateUploadPromises.push(generatedCertificateResponse);
+        } else {
+            certificateUploadPromises.push(Promise.reject(new Error("Record does not have valid testResultId for certificate generation")));
```
Comment (acozianu) [inline src/functions/certGen.ts]: Please add the string in the enum file

Comment (acozianu) [reply src/functions/certGen.ts]: done

File: src/functions/certGen.ts (hunk 2 above)
Comment (acozianu) [reply src/functions/certGen.ts]: done

**Pull request merged**